.release-please-manifest.json

```
@@ -1,3 +1,3 @@
 {
-    "hugr-py": "0.14.2"
+    "hugr-py": "0.14.4"
```

release-please uses this to decide which commits to consider, looking at the release tag.

This patch tag is out-of-tree, so the automation wouldn't be able to find the right range.

We should leave it at `0.14.2`, and remove the already-picked changelog lines manually.

```suggestion
    "hugr-py": "0.14.2"
```